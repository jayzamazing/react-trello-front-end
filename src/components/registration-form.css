.block {
  display: block;
}
.input-field {
  width: 340px;
}
.inline-block {
  display: inline-block;
}
.space {
  margin: 20px 5px 0px 0px;
}
