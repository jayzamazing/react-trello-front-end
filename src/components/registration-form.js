import React from 'react';
import {Field, reduxForm, focus} from 'redux-form';
import {required, nonEmpty, length, isTrimmed} from '../validators';
import Input from './input';
import Checkbox from './checkbox';
import {createUser} from '../actions/users';
import {login} from '../actions/auth';
import './registration-form.css';

export class RegistrationForm extends React.Component {
  onSubmit(values) {
      const {email, password, fullName} = values;
      const user = {email, password, fullName};
      return this.props
          .dispatch(createUser(user))
          .then(() => this.props.dispatch(login(email, password)));
  }
  render() {
    return (
        <form
            className="login-form"
            onSubmit={this.props.handleSubmit(values =>
                this.onSubmit(values)
            )}>
            <Field
              label="Name"
              component={Input}
              type="text"
              name="fullName"
              validate={[required, nonEmpty, isTrimmed]}
              inputClass="input-field form-control"
            />
            <Field
              label="Email"
              component={Input}
              type="text"
              name="email"
              validate={[required, nonEmpty, isTrimmed]}
              inputClass="input-field form-control"
            />
            <Field
              label="Password"
              component={Input}
              type="password"
              name="password"
              validate={[required, length({min: 10, max: 72}), isTrimmed]}
              inputClass="input-field form-control"
            />
            <Field
              label={["I accept the ", <a href="#" key="0">Terms of Service</a>," and ", <a href="#" key="1">Privacy Policy</a>]}
              component={Checkbox}
              type="checkbox"
              name="acceptTerms"
              validate={[required]}
              divClass="inline-block space"
              inputClass="checkbox-space"
            />

          <button className="block btn btn-success"
              type="submit"
              disabled={this.props.pristine || this.props.submitting}>
              Create New Account
          </button>
        </form>
    );
  }
}

export default reduxForm({
  form: 'registration',
  onSubmitFail: (errors, dispatch) =>
      dispatch(focus('login-form', Object.keys(errors)[0]))
})(RegistrationForm);
