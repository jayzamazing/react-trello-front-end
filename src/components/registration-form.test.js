'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {RegistrationForm} from './registration-form';

describe('login-form', () => {
  //ensure that it renders without issues
  it('should render', () => {
    const handleSubmit = jest.fn();
    shallow(<RegistrationForm handleSubmit={handleSubmit}/>);
  });
  //take snapshot for future changes
  it('snapshot', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<RegistrationForm handleSubmit={handleSubmit}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on login button and check dispatches
  it('login button click', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<RegistrationForm handleSubmit={handleSubmit}/>);
    wrapper.find('[name="fullName"]').at(0).simulate('change', { target: { value: 'Super Man' } });
    wrapper.find('[name="email"]').at(0).simulate('change', { target: { value: 'test@test.com' } });
    wrapper.find('[name="password"]').at(0).simulate('change', { target: { value: 'somereallylongpassword2343' } });
    wrapper.find('[name="acceptTerms"]').at(0).simulate('change', {target: {checked: true}});
    wrapper.find('.btn-success').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
});
