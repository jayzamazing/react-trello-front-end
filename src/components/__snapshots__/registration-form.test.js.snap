// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login-form snapshot 1`] = `
<form
  className="login-form"
>
  <Field
    component={[Function]}
    inputClass="input-field form-control"
    label="Name"
    name="fullName"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    inputClass="input-field form-control"
    label="Email"
    name="email"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    inputClass="input-field form-control"
    label="Password"
    name="password"
    type="password"
    validate={
      Array [
        [Function],
        [Function],
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    divClass="inline-block space"
    inputClass="checkbox-space"
    label={
      Array [
        "I accept the Terms of Service, and Privacy Policy.",
      ]
    }
    labelclass="checkbox-label"
    name="acceptTerms"
    type="checkbox"
    validate={
      Array [
        [Function],
      ]
    }
  />
  <button
    className="block btn btn-success create-acc-btn"
    type="submit"
  >
    Create New Account
  </button>
</form>
`;
